src/result/mod.rs

```diff
@@ -80,25 +81,25 @@ pub(crate) trait IonFailure {
     // Because this trait is only crate-visible, methods can be added/changed as needed in
     // the future.
     fn incomplete(label: &'static str, position: impl Into<Position>) -> Self;
-    fn decoding_error<S: Into<String>>(description: S) -> Self;
-    fn encoding_error<S: Into<String>>(description: S) -> Self;
-    fn illegal_operation<S: Into<String>>(operation: S) -> Self;
+    fn decoding_error<S: Into<Cow<'static, str>>>(description: S) -> Self;
```

Comment:
Thinking out loud: Is it possible to typedef `Cow<'static, str>` or something? Is that even helpful?

Reply:
Yes! I want to see this:
```rust
type Moo = Cow<'static, str>;
```
 😉